File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/EntityLookPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.server.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public record EntityLookPacket(int entityId, byte yaw, byte pitch) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        os.writeByte(yaw);
        os.writeByte(pitch);
    }

    @Override
    public byte getId() {
        return 0x16;
    }
}

File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/EntityLookPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class EntityLookPacket(val entityId: Int, val yaw: Byte, val pitch: Byte) : Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId)
        os.writeByte(yaw.toInt())
        os.writeByte(pitch.toInt())
    }

    override fun getId(): Byte {
        return 0x16
    }
}
